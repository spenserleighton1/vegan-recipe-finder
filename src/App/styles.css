.App {
  text-align: center;
  width: 90vw;
  padding: 50px;
}
