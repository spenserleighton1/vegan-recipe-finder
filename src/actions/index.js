export const addRecipes = (recipes) => ({
  type: 'ADD_RECIPES',
  recipes
})

export const searchByIngredient = (ingredients) => ({
  type: 'SEARCH_BY_INGREDIENT', 
  ingredients
})