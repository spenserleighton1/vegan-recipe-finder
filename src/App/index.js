import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, withRouter } from 'react-router-dom';
import './styles.css';
import { fetchRecipes } from '../helper/apiCalls';
import { apiKey } from '../helper/apiKey';
import { cleanData } from '../helper/dataCleaner';
import { addRecipes } from '../actions';
import RecipesContainer from '../RecipesContainer';
import Header from '../Header'
import Search from '../Search'

class App extends Component {

async componentDidMount() {
    const results = await fetchRecipes(apiKey);
    const recipes = await cleanData(results.recipes);
    this.props.addRecipes(recipes)
  }

  render() {
    return (
      <div className="App">
      <Header />
      <Search />
      <RecipesContainer />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addRecipes: (recipes)=> dispatch(addRecipes(recipes))
})

export default withRouter(connect(null, mapDispatchToProps)(App));
