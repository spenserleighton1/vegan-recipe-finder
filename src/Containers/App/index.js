import { addRecipes, isLoading } from '../../actions';
import SingleRecipeContainer from '../../Components/SingleRecipeContainer';
import SavedRecipesContainer from '../SavedRecipesContainer';
import { Route, withRouter } from 'react-router-dom';
import React, { Component } from 'react';
import { firebase } from '../../firebase';
import RecipesContainer from '../RecipesContainer';
import { fetchRecipes } from '../../helper/apiCalls';
import { cleanData } from '../../helper/dataCleaner';
import { connect } from 'react-redux';
import { apiKey } from '../../helper/apiKey';
import SignUpPage from '../../Components/SignUp';
import SignInPage from '../../Components/SignIn';
import Header from '../../Components/Header';
import Loader from '../../Components/Loader';
import PropTypes from 'prop-types';

import './styles.css';


export class App extends Component {
  constructor(props) {
    super(props);
    
    this.state = { authUser: null };
  }

  async componentDidMount() {
    this.props.isLoading(true);
    firebase.auth.onAuthStateChanged(authUser => {
      authUser
        ? this.setState({ authUser })
        : this.setState({ authUser: null });
    });

    const results = await fetchRecipes(apiKey);
    const recipes = await cleanData(results.recipes);
    this.props.addRecipes(recipes);
    await this.props.isLoading(false);
  }

  render() {
    const loaded = this.props.loading ? <Loader /> && <RecipesContainer /> : <RecipesContainer />;
    return (
      <div className="App">
        <Header authUser={ this.state.authUser } />
        <Route exact path={'/signUp'} component={SignUpPage} />
        <Route exact path={'/signIn'} component={SignInPage} />
        <Route exact path={'/savedRecipes'} render={() => <SavedRecipesContainer authUser={ this.state.authUser } />} />
        <Route exact path={'/'} render={()=> <SingleRecipeContainer authUser={ this.state.authUser } /> } />
        <Route exact path={'/'} render={()=> loaded } />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  loading: state.loading
});

export const mapDispatchToProps = (dispatch) => ({
  isLoading: (bool) => dispatch(isLoading(bool)),
  addRecipes: (recipes)=> dispatch(addRecipes(recipes))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));

App.propTypes = {
  isLoading: PropTypes.func.isRequired,
  addRecipes: PropTypes.func.isRequired,
  loading: PropTypes.bool
};
