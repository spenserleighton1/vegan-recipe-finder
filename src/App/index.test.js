import React from 'react';
import { shallow } from 'enzyme';
import App, { mapDispatchToProps } from './';
import { addRecipes } from '../actions';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot()
  })
  describe('mapDispatchToProps', () => {
    it('should return propsObject', () => {
      const mockAddRecipes = jest.fn();
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = addRecipes([{title: 'food'}])

      mappedProps.addRecipes([{title: 'food'}])
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
