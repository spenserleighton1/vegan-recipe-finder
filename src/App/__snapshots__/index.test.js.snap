// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addRecipes={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          authUser={null}
/>,
        <Connect(Search) />,
        <Connect(SingleRecipeContainer) />,
        <Connect(RecipesContainer) />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signUp"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signIn"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/passwordForgetPage"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/home"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/account"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authUser": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signUp",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signIn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/passwordForgetPage",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/account",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            authUser={null}
/>,
          <Connect(Search) />,
          <Connect(SingleRecipeContainer) />,
          <Connect(RecipesContainer) />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signUp"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signIn"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/passwordForgetPage"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/home"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/account"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authUser": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signUp",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signIn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/passwordForgetPage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/account",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
